src/main/java/com/cvsgo/repository/ProductCustomRepositoryImpl.java

```diff
@@ -58,72 +69,142 @@ public List<SearchProductQueryDto> searchByFilter(User user,
+                            priceGreaterOrEqual(searchFilter.getLowestPrice())
+                        ))
+            )
+            .fetchFirst();
+    }
+
     public List<ConvenienceStoreEventQueryDto> findConvenienceStoreEventsByProductIds(
         List<Long> productIds) {
         return queryFactory
             .select(new QConvenienceStoreEventQueryDto(
                 sellAt.product.id,
                 sellAt.convenienceStore.name,
-                event.eventType))
+                event))
             .from(sellAt)
             .leftJoin(event).on(sellAt.product.eq(event.product)
                 .and(sellAt.convenienceStore.eq(event.convenienceStore)))
             .where(sellAt.product.id.in(productIds))
             .fetch();
     }
 
-    public Optional<ProductDetailResponseDto> findByProductId(User user, Long productId) {
+    public Optional<SearchProductDetailQueryDto> findByProductId(User loginUser, Long productId) {
         return Optional.ofNullable(queryFactory
-            .select(new QProductDetailResponseDto(
-                product.as("product"),
-                product.manufacturer.as("manufacturer"),
-                productLike.as("productLike"),
-                productBookmark.as("productBookmark")
+            .select(new QSearchProductDetailQueryDto(
+                product.id,
+                product.name,
+                product.price,
+                product.imageUrl,
+                manufacturer.name,
+                productLike,
+                productBookmark
             ))
             .from(product)
             .leftJoin(productLike)
-            .on(productLike.product.id.eq(productId).and(productLike.user.eq(user)))
+            .on(productLike.product.eq(product).and(productLikeUserEq(loginUser)))
             .leftJoin(productBookmark)
-            .on(productBookmark.product.id.eq(productId).and(productBookmark.user.eq(user)))
+            .on(productBookmark.product.eq(product).and(productBookmarkUserEq(loginUser)))
+            .leftJoin(manufacturer).on(product.manufacturer.eq(manufacturer))
             .where(product.id.eq(productId))
             .fetchFirst());
     }
 
-    private BooleanExpression eqProductLikeUser(User user) {
+    public List<ConvenienceStoreEventQueryDto> findConvenienceStoreEventsByProductId(
+        Long productId) {
+        return queryFactory.select(new QConvenienceStoreEventQueryDto(
+                sellAt.convenienceStore.id,
+                sellAt.convenienceStore.name,
+                event))
+            .from(sellAt)
+            .leftJoin(event).on(sellAt.product.eq(event.product)
+                .and(sellAt.convenienceStore.eq(event.convenienceStore)))
+            .leftJoin(convenienceStore).on(sellAt.convenienceStore.eq(convenienceStore))
+            .where(sellAt.product.id.eq(productId))
+            .fetch();
+    }
+
+    private static List<OrderSpecifier> sortBy(ProductSortBy sortBy, NumberPath<Double> score,
```

Comment:
OrderSpecifier가 raw 타입이라 이렇게 wildcard를 사용하는 것이 좋을 것 같습니다!
```suggestion
    private static List<OrderSpecifier<?>> sortBy(ProductSortBy sortBy, NumberPath<Double> score,
```